packages/m2m-gateway/test/api/eServiceTemplates/updateDraftEServiceTemplateVersion.test.ts

```diff
@@ -0,0 +1,172 @@
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+import { describe, it, expect, vi } from "vitest";
+import request from "supertest";
+import { generateId, pollingMaxRetriesExceeded } from "pagopa-interop-models";
+import { AuthRole, authRole } from "pagopa-interop-commons";
+import { m2mGatewayApi } from "pagopa-interop-api-clients";
+import {
+  generateToken,
+  getMockedApiEserviceTemplateVersion,
+} from "pagopa-interop-commons-test";
+import { api, mockEServiceTemplateService } from "../../vitest.api.setup.js";
+import { toM2MGatewayEServiceTemplateVersion } from "../../../src/api/eserviceTemplateApiConverter.js";
+import {
+  eserviceTemplateVersionNotFound,
+  missingMetadata,
+} from "../../../src/model/errors.js";
+import { appBasePath } from "../../../src/config/appBasePath.js";
+import { config } from "../../../src/config/config.js";
+
+describe("PATCH /eserviceTemplates/:templateId/versions/:versionId/ router test", () => {
+  const versionSeed = {
+    description: "Test Version Description",
+    voucherLifespan: 100,
+    dailyCallsPerConsumer: 10,
+    dailyCallsTotal: 10,
+    agreementApprovalPolicy: "AUTOMATIC",
+  };
+
+  const mockM2MTemplateVersionResponse: m2mGatewayApi.EServiceTemplateVersion =
+    toM2MGatewayEServiceTemplateVersion(getMockedApiEserviceTemplateVersion());
+
+  mockEServiceTemplateService.updateDraftEServiceTemplateVersion = vi
+    .fn()
+    .mockResolvedValue(mockM2MTemplateVersionResponse);
+
+  const makeRequest = async (
+    token: string,
+    templateId: string = generateId(),
+    versionId: string = generateId(),
+    body: typeof versionSeed = versionSeed
+  ) =>
+    request(api)
+      .patch(
+        `${appBasePath}/eserviceTemplates/${templateId}/versions/${versionId}`
+      )
+      .set("Authorization", `Bearer ${token}`)
+      .send(body);
+
+  const authorizedRoles: AuthRole[] = [authRole.M2M_ADMIN_ROLE];
+  it.each(authorizedRoles)(
+    "Should return 200 for user with role %s",
+    async (role) => {
+      const token = generateToken(role);
+      const res = await makeRequest(token);
+      expect(res.status).toBe(200);
+      expect(res.body).toEqual(mockM2MTemplateVersionResponse);
+    }
+  );
+
+  it.each(
+    Object.values(authRole).filter((role) => !authorizedRoles.includes(role))
+  )("Should return 403 for user with role %s", async (role) => {
+    const token = generateToken(role);
+    const res = await makeRequest(token);
+
+    expect(res.status).toBe(403);
+  });
+
+  it.each([
+    {},
+    {
+      description: "Updated Description",
+    },
+    {
+      description: "Another Updated Description",
+      voucherLifespan: 300,
+    },
+    {
+      description: "Yet Another Updated Description",
+      voucherLifespan: 300,
+      dailyCallsPerConsumer: 30,
+      dailyCallsTotal: 30,
+    },
+  ] as Array<Partial<typeof versionSeed>>)(
+    "Should return 200 with partial seed (seed #%#)",
```

Comment:
Same here as comments above
```suggestion
    "Should return 200 with partial seed and nullable fields (seed #%#)",
```